dynet/lstm.h

```diff
@@ -317,6 +318,157 @@ struct VanillaLSTMBuilder : public RNNBuilder {
+                              ParameterCollection& model,
+                              bool ln_lstm = false,
+                              float forget_bias = 1.f);
+
+  Expression back() const override { return (cur == -1 ? h0.back() : h[cur].back()); }
+  std::vector<Expression> final_h() const override { return (h.size() == 0 ? h0 : h.back()); }
+  std::vector<Expression> final_s() const override {
+    std::vector<Expression> ret = (c.size() == 0 ? c0 : c.back());
+    for (auto my_h : final_h()) ret.push_back(my_h);
+    return ret;
+  }
+  unsigned num_h0_components() const override { return 2 * layers; }
+
+  std::vector<Expression> get_h(RNNPointer i) const override { return (i == -1 ? h0 : h[i]); }
+  std::vector<Expression> get_s(RNNPointer i) const override {
+    std::vector<Expression> ret = (i == -1 ? c0 : c[i]);
+    for (auto my_h : get_h(i)) ret.push_back(my_h);
+    return ret;
+  }
+
+  void copy(const RNNBuilder & params) override;
+
+  /**
+   * \brief Set the dropout rates to a unique value
+   * \details This has the same effect as `set_dropout(d,d_h)` except that all the dropout rates are set to the same value.
+   * \param d Dropout rate to be applied on all of \f$x,h\f$
+   */
+  virtual void set_dropout(float d) override;
+  /**
+   * \brief Set the dropout rates
+   * \details The dropout implemented here is the variational dropout with tied weights introduced in [Gal, 2016](http://papers.nips.cc/paper/6241-a-theoretically-grounded-application-of-dropout-in-recurrent-neural-networks)
+   * More specifically, dropout masks \f$\mathbf{z_x}\sim \mathrm{Bernoulli}(1-d_x)\f$,\f$\mathbf{z_h}\sim \mathrm{Bernoulli}(1-d_h)\f$ are sampled at the start of each sequence.
+   * The dynamics of the cell are then modified to :
+   *
+   * \f$
+   * \begin{split}
+    i_t & =\sigma(W_{ix}(\frac 1 {1-d_x}\mathbf{z_x} \circ x_t)+W_{ih}(\frac 1 {1-d_h}\mathbf{z_h} \circ h_{t-1})+b_i)\\
+    f_t & = \sigma(W_{fx}(\frac 1 {1-d_x}\mathbf{z_x} \circ x_t)+W_{fh}(\frac 1 {1-d_h}\mathbf{z_h} \circ h_{t-1})+b_f)\\
+    o_t & = \sigma(W_{ox}(\frac 1 {1-d_x}\mathbf{z_x} \circ x_t)+W_{oh}(\frac 1 {1-d_h}\mathbf{z_h} \circ h_{t-1})+b_o)\\
+    \tilde{c_t} & = \tanh(W_{cx}(\frac 1 {1-d_x}\mathbf{z_x} \circ x_t)+W_{ch}(\frac 1 {1-d_h}\mathbf{z_h} \circ h_{t-1})+b_c)\\
+    c_t & = c_{t-1}\circ f_t + \tilde{c_t}\circ i_t\\
+    h_t & = \tanh(c_t)\circ o_t\\
+   \end{split}
+   * \f$
+   *
+   * For more detail as to why scaling is applied, see the "Unorthodox" section of the documentation
+   * \param d Dropout rate \f$d_x\f$ for the input \f$x_t\f$
+   * \param d_h Dropout rate \f$d_h\f$ for the output \f$h_t\f$
+   */
+  void set_dropout(float d, float d_r);
+  /**
+   * \brief Set all dropout rates to 0
+   * \details This is equivalent to `set_dropout(0)` or `set_dropout(0,0,0)`
+   *
+   */
+   
+   void set_sparsity(float percent);
+   /**
+   * \brief Select the percent of weights that will be masked, based on their magnitude
+   */
```

Comment:
I think the docstrings for the two methods here (set_sparsity and disable_dropout) are interchanged.